python/pyspark/testing/__init__.py

```diff
@@ -16,6 +16,11 @@
 #
 from pyspark.testing.utils import assertDataFrameEqual, assertSchemaEqual
 
-from pyspark.testing.pandasutils import assertPandasOnSparkEqual
+__all__ = ["assertDataFrameEqual", "assertSchemaEqual"]
 
-__all__ = ["assertDataFrameEqual", "assertSchemaEqual", "assertPandasOnSparkEqual"]
+try:
+    from pyspark.testing.pandasutils import assertPandasOnSparkEqual
```

Comment:
I think you should still keep the API, and throw an exception within this API.

Reply:
Moved `assertPandasOnSparkEqual` from `pyspark.testing.pandasutils` to `pyspark.testing.psutils` to remove pandas dependency from the existing testing utils keeping `assertPandasOnSparkEqual` as an API as is.